std/algebra/emulated/sw_emulated/hints.go

```diff
@@ -70,42 +93,362 @@ func decomposeScalarG1Signs(mod *big.Int, inputs []*big.Int, outputs []*big.Int)
 	})
 }
 
-func decomposeScalarG1(mod *big.Int, inputs []*big.Int, outputs []*big.Int) error {
+// TODO @yelhousni: generalize for any supported curve.
+// as it currently works only for P-256, P-384 and STARK curve.
+func scalarMulG1Hint(_ *big.Int, inputs []*big.Int, outputs []*big.Int) error {
+	return emulated.UnwrapHintWithNativeInput(inputs, outputs, func(field *big.Int, inputs, outputs []*big.Int) error {
+		if len(outputs) != 2 {
+			return fmt.Errorf("expecting two outputs")
+		}
+		if field.Cmp(elliptic.P256().Params().P) == 0 {
+			var fp emparams.P256Fp
+			var fr emparams.P256Fr
+			PXLimbs := inputs[:fp.NbLimbs()]
+			PYLimbs := inputs[fp.NbLimbs() : 2*fp.NbLimbs()]
+			SLimbs := inputs[2*fp.NbLimbs():]
+			Px, Py, S := new(big.Int), new(big.Int), new(big.Int)
+			if err := limbs.Recompose(PXLimbs, fp.BitsPerLimb(), Px); err != nil {
+				return err
+
+			}
+			if err := limbs.Recompose(PYLimbs, fp.BitsPerLimb(), Py); err != nil {
+				return err
+
+			}
+			if err := limbs.Recompose(SLimbs, fr.BitsPerLimb(), S); err != nil {
+				return err
+
+			}
+			curve := elliptic.P256()
+			// compute the resulting point [s]Q
+			outputs[0], outputs[1] = curve.ScalarMult(Px, Py, S.Bytes())
```

Comment:
Instead of assigning to `outputs[0]` and `outputs[1]`, you should obtain the values from the `ScalarMult` and then do `outputs[0].Set()` and `outputs[1].Set`. We use `*big.Int` pool and imo would otherwise leak.

Similarly for below.